db/sqloffload.c

```diff
@@ -746,6 +746,10 @@ static void osql_scdone_commit_callback(struct ireq *iq)
     int write_scdone =
         bdb_attr_get(thedb->bdb_attr, BDB_ATTR_SC_DONE_SAME_TRAN) ? 0 : 1;
     gbl_readonly_sc = 0;
+    if (iq->sc_locked) {
+        unlock_schema_lk();
+        iq->sc_locked = 0;
+    }
```

Comment:
Don't understand why this is ok to move up to here. Will another sc be able to sneak in and potentially cause corruption if we release schema_lk here?

Reply:
This is a commit callback function and by the time we are here, the schema change has already committed.